## Cars Data Analysis by Chidi Ikebude.
-----

In [1]:
import pandas as pd

In [2]:
import numpy as np

In [4]:
!pip install nbconvert

In [1]:
import plotly.io as pio
pio.renderers.default = 'notebook'

In [12]:
pip install nbconvert>=6.4.4

Note: you may need to restart the kernel to use updated packages.


In [13]:
pip install jedi>=0.17.2

Note: you may need to restart the kernel to use updated packages.


## Working on Real Project With Python
The Cars Dataset

In [3]:
cars = pd.read_csv(r"C:\Users\user\Projects\Data-Analysis-Projects\Cars Data Analysis\CarsDataset.csv")

In [4]:
cars

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[428 rows x 15 columns]

## How to Analyze DataFrames
Lets look at the description of the dataframe

In [5]:
cars.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

Lets look at the total number of rows and Columns

In [6]:
cars.shape

(428, 15)

To look at all the codes information about the whole data feame we use .info

In [7]:
cars.info

<bound method DataFrame.info of       Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Hig

Now lets dive into analyzing the data in full

## Lets run some Analysis to view some stats.

 1) Instruction (For Data Cleaning)
- Find all Null Values in the dataset. If there is any null value in any column, then fill it with the mean of that column.

In [8]:
cars.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      2
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

There are 2 null values in the Cylinders column, lets fix that up with the mean of the column Cylinders

In [9]:
cars['Cylinders'].fillna(cars['Cylinders'].mean(), inplace = True)

The `inplace = TRUE` makes the code be applicable to the column in the dataframe

2) Question ( Based on Value Counts )
- Check what are the different types of Make are there in our dataset. And, what is the count (occurrence) of each Make in the data

In [10]:
cars.Make.value_counts()

Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Honda            17
Nissan           17
Volkswagen       15
Chrysler         15
Dodge            13
Mitsubishi       13
Volvo            12
Jaguar           12
Hyundai          12
Subaru           11
Pontiac          11
Mazda            11
Lexus            11
Kia              11
Buick             9
Mercury           9
Lincoln           9
Saturn            8
Cadillac          8
Suzuki            8
Infiniti          8
GMC               8
Acura             7
Porsche           7
Saab              7
Land Rover        3
Oldsmobile        3
Jeep              3
Scion             2
Isuzu             2
MINI              2
Hummer            1
Name: Make, dtype: int64

Toyota has the highest make with 27

3) Instruction ( Filtering )
- Show all the records where Origin is Asia or Europe.

In [11]:
cars[(cars.Origin == 'Asia') | (cars.Origin == 'Europe')]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[281 rows x 15 columns]

Total of 281  Rows. we could use another function

In [12]:
cars[cars['Origin'].isin(['Asia', 'Europe'])]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[281 rows x 15 columns]

They both bring the same data of 281 rows

4) Instruction ( Removing unwanted records )
- Remove all the records (rows) where Weight is above 4000.

In [13]:
cars[~(cars['Weight'] > 4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
5    $41,100          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
5      3893        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[325 rows x 15 columns]

That could work. `cars[cars.Weight <= 4000]` could also work

In [21]:
cars[(cars['Weight'] <= 4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
5    $41,100          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
5      3893        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[325 rows x 15 columns]

5) Instruction ( Applying function on a column )
- Increase all the values of 'MPG_City' column by 3.

In [22]:
cars['MPG_City'] =  cars['MPG_City'].apply(lambda x:x+3)

In [23]:
cars

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        20           23   
1    $21,761          2.0        4.0         200        27           31   
2    $24,647          2.4        4.0         200        25           29   
3    $30,299          3.2        6.0         270        23           28   
4    $39,014          3.5        6.0         225        21           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        24           28   
424  $40,083          2.3        5.0         242        23           26   
425  $42,573          2.9        6.0         268        22           26   
426  $24,641          1.9        4.0         170        25           29   
427  $33,112          2.5        5.0         208        23           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[428 rows x 15 columns]

----